### ISSUE TITLE
AMO add-on/extension search results at (in) about:addons

### ISSUE THREAD

**Issue by grahamperrin** [BrowserWorks/waterfox#873]
![image](https://user-images.githubusercontent.com/192271/53656870-7f27b300-3c4c-11e9-99d4-b9bce070ff78.png)

– _Could not find any matching add-ons_

From IRC:

>> Re: https://www.reddit.com/r/waterfox/comments/atwjys/-/ehhf79p/ might a v3 or v4 API string allow search results to be returned in ye olde about:addons GUI?

> yes, but it's not just the url that's changed - the developers will need to update the code to handle the newer v3+ api https://addons-server.readthedocs.io/en/latest/topics/api/index.html
> (or https://addons-server.readthedocs.io/en/latest/topics/api/v3_legacy/index.html#v3-api-index)

@MrAlex94 I'm closing this, assuming _wontfix_ but please reopen if you feel the need for the enhancement.

**Comment by ticalc-travis** [comment]
If that search interface can no longer work, then it should probably should be removed or an alternative provided (such as a link to the add-ons site). I was trying to use it just now and couldn't figure out what was going on. From a user perspective, that's confusing.

**Comment by MrAlex94** [comment]
Good shout. Should replace it with the classic add-on archive :-)

**Comment by grahamperrin** [comment]
@ticalc-travis true. Sorry. My _wontfix_ assumption was based on available (human) resources for other interfaces that mislead the end user: 

- [about:addons 'Check to see if your plugins are up to date' · Issue #336 · MrAlex94/Waterfox](https://github.com/MrAlex94/Waterfox/issues/336)
- [Waterfox ▸ Help ▸ Waterfox help · Issue #400 …](https://github.com/MrAlex94/Waterfox/issues/400) partially duplicated by #548 with a non-merged pull request at #573
- [Non-functional 'Enable' button and misleading yellow alert at about:addons · Issue #878 …](https://github.com/MrAlex94/Waterfox/issues/878)

… like, I'm a human resource but if I were to attempt pull requests for issues such as those, I'd make a _horrible_ mess `:-/`

----

@MrAlex94 aiming to return CAA results at which of these two addresses? 

1. `about:addons` in the same tab
2. `caa:list?q=`_search string_ in a new tab

Either way, the find interface (prior to gaining results) should make clear that it'll be an unusual search. Even if you make it explicit, expect some users to use the search box without paying attention; risk of ending up with an outdated extension without the user realising that there's a more recent version. 

FWIW I prefer either of the other options, 

> … removed or … a link to the add-ons site. …

– and for the latter, the link should be non-localised https://addons.mozilla.org/

Note, the caa: reference to the live site is localised and IIRC (I can't find the relevant issue) will not be de-localised. 

Meta/tracking: #538

**Comment by ticalc-travis** [comment]
Yeah, when I first set up Waterfox last month (migrating from the last FF ESR that still supported legacy extensions) I opted to create a fresh, new profile and might have initially just gone straight to the AMO site to download some current add-ons and then found the classic add-on archive and grabbed all the legacy stuff I needed. :-) It's possible I never tried the search box until today when I got around to recreating my alternate profile from ESR. Was puzzling, though, since I was expecting *something* to come up from AMO and had no idea if it ever worked before.

(BTW, I'm loving the huge performance improvement in WF over that ESR version, even after adding all my stuff back. ESR 52 was getting to where it was constantly lagging for no apparent reason, sometimes up to a minute at a time, and I was getting sick of using it.)
